File: AppointmentBooking.Server/Models/Appointment.cs
namespace AppointmentBooking.Server.Models;

public class Appointment
{
    public int Id { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public AppointmentStatus Status { get; set; }
    public string? Notes { get; set; }
    public required int ServiceProviderId { get; set; }
    public required ServiceProvider ServiceProvider { get; set; }
    public required string UserId { get; set; }
    public ApplicationUser User { get; set; } = null!;
    public required int ServiceId { get; set; }
    public required Service Service { get; set; }
}

File: appointmentbooking.client/src/types/Appointment.ts
export interface AvailableTimeDto {
  startTime: string;
  endTime: string;
  displayTime: string;
}

export interface CreateAppointmentDto {
  startTime: string;
  serviceProviderId: number;
  serviceId: number;
  notes: string;
}

export interface AppointmentBookingProps {
  serviceProviderId: number;
  serviceId: number;
  serviceName: string;
  serviceDuration?: number;
  servicePrice?: number;
  onBack?: () => void;
  onSuccess?: () => void;
}

export interface Appointment {
  id: number;
  startTime: string;
  endTime: string;
  userId: string;
  serviceProviderId: string;
  serviceId: string;
  status: string;
  notes: string;
  serviceName: string;
  serviceProviderName: string;
}
